src/refactorings/convert-let-to-const/convert-let-to-const.test.ts

```diff
@@ -0,0 +1,160 @@
+import { Editor, ErrorReason, Code } from "../../editor/editor";
+import { Selection } from "../../editor/selection";
+import { InMemoryEditor } from "../../editor/adapters/in-memory-editor";
+import { testEach } from "../../tests-helpers";
+
+import { convertLetToConst } from "./convert-let-to-const";
+
+describe("Convert Let To Const", () => {
+  let showErrorMessage: Editor["showError"];
+
+  beforeEach(() => {
+    showErrorMessage = jest.fn();
+  });
+
+  testEach<{ code: Code; selection?: Selection; expected: Code }>(
+    "should convert let to const",
+    [
+      {
+        description: "non-mutated variable declared not in a block",
+        code: `let someVariable = 'value';`,
+        expected: `const someVariable = 'value';`
+      },
+      {
+        description: "non-mutated variable declared in a block",
+        code: `{
+  let someVariable = 'value';
+}`,
+        expected: `{
+  const someVariable = 'value';
+}`
+      },
+      {
+        description: "multiple non-mutated variables declared together",
+        code: `{
+  let someVariable, otherVariable = 'value';
+}`,
+        expected: `{
+  const someVariable, otherVariable = 'value';
+}`
+      },
+      {
+        description: "multiple non-mutated variables delcared seperately",
+        code: `{
+  let someVariable = 'someValue';
+  let otherVariable = 'otherValue';
+}`,
+        expected: `{
+  const someVariable = 'someValue';
+  let otherVariable = 'otherValue';
+}`
+      },
+      {
+        description:
+          "multiple variables delcared seperately, other one mutated",
```

Comment:
yes that makes sense, my bad, I'd interpreted the selection to have a wider scope (i.e. all variables on that line) than it does, rather than actually a single selected variable.

I agree with your thoughts so will make the changes 👍